Cabal/src/Distribution/Simple/Utils/Json.hs

```diff
@@ -1,50 +0,0 @@
--- | Extremely simple JSON helper. Don't do anything too fancy with this!
```

Comment:
Would have been nice to do the refactor and move of this module in separate commits to more easily see what changed.

Reply:
The json changes have been extracted to https://github.com/haskell/cabal/pull/7477, this PR will be rebased on top of it once it has been merged.

Reply:
Changes are merged independently!